packages/rum/src/domain/record/serializationUtils.ts

```diff
@@ -1,14 +1,13 @@
 import { buildUrl } from '@datadog/browser-core'
 import { CENSORED_STRING_MARK, NodePrivacyLevel } from '../../constants'
 import { shouldMaskNode } from './privacy'
-import { SerializedNodeWithId } from './types'
 
-export interface NodeWithSerializedNode extends Node {
-  __sn: SerializedNodeWithId
-}
+export type NodeWithSerializedNode = Node & { s: 'Node with serialized node' }
+
+const serializedNodeIds = new WeakMap<Node, number>()
```

Comment:
what is the reasoning of using a `WeakMap` rather than continue storing the id directly on the node?

Reply:
We avoid setting a visible property in our customers app. Also it solves issues when the SDK is instanciated twice. And in my mind it's simply more elegent.

Do you see something wrong with this?

Reply:
no, I was just wondering :sli

Reply:
Yes, I also think it's a great use of it :)